tests/m2m_through/tests.py

```diff
@@ -20,6 +20,27 @@ def setUpTestData(cls):
         cls.rock = Group.objects.create(name='Rock')
         cls.roll = Group.objects.create(name='Roll')
 
+    # #32947
+    def test_through_fields_as_list_are_hashable_in_proxy_parent_model(self):
+        # no error raised
+        PersonProxy._check_field_name_clashes()
+
+        proxy_parents = PersonProxy._meta.get_parent_list()
+        self.assertEqual(proxy_parents, [Person])
+
+        events_invited_fields = [
+            f for f in Person._meta.get_fields()
+            if f.name == 'events_invited'
+        ]
+        self.assertEqual(len(events_invited_fields), 1)
+        events_invited_field = events_invited_fields[0]
+
+        self.assertIn(('event', 'invitee'), events_invited_field.identity)
+        self.assertEqual(['event', 'invitee'], events_invited_field.through_fields)
+
+        # not rising error
+        events_invited_field.__hash__()
```

Comment:
TIL about `get_field` method. 

Though I think it should also need to make sure that identity is a type of tuple. Otherwise, someone in the future might change the type of through fields in the test model and this test case will stop testing the very thing it's supposed to test.